### ISSUE TITLE
Issues while using GraphSage layers

### ISSUE THREAD

**Issue by VikashPeddakota999** [tensorflow/gnn#311]
TryinG to use the following layers: https://github.com/tensorflow/gnn/blob/main/tensorflow_gnn/models/graph_sage/layers.py

Getting the following errors

While compilig the model, 
**WARNING:tensorflow:AutoGraph could not transform <function check_scalar_graph_piece at 0x7f70028b4160> and will run it as-is.**

While calling model.fit() its showing the following warnings:
**WARNING:tensorflow:Gradients do not exist for variables ['graph_update/node_set_update_1/graph_sage_aggregator_conv/dense_3/kernel:0', 'graph_update/node_set_update_1/graph_sage_next_state_1/bias:0', 'graph_update/node_set_update_1/graph_sage_next_state_1/dense_4/kernel:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss` argument?**

**Comment by phanein** [comment]
Hi @VikashPeddakota999,

Thanks for reporting these.   If you have a publicly available colab notebook that reproduces this error, we might have more comments.  At a glance, it really does seem like your graph update is miswired.  Are you using [GraphSAGEGraphUpdate](https://github.com/tensorflow/gnn/blob/ad0ee136e3a131771e313d9092d1295270bc30c9/tensorflow_gnn/models/graph_sage/layers.py#L477)?  (the name 'node_set_update_1' makes me think you may not be)

But for now, just some basic comments about these messages for the record --

I'm not too concerned about this:

> WARNING:tensorflow:AutoGraph could not transform <function check_scalar_graph_piece at 0x7f70028b4160> and will run it as-is.

However, errors like these are more interesting, and imply that something isn't fully hooked up in your model.  In this case I can only speculate, but it seems quite possible that you could have added some transformations over features that are not used later in your model.

> WARNING:tensorflow:Gradients do not exist for variables ['graph_update/node_set_update_1/graph_sage_aggregator_conv/dense_3/kernel:0', 'graph_update/node_set_update_1/graph_sage_next_state_1/bias:0', 'graph_update/node_set_update_1/graph_sage_next_state_1/dense_4/kernel:0']


For instance, if I modify the [MUTAG tutorial](https://github.com/tensorflow/gnn/blob/main/examples/tutorials/neurips_2022/neurips_student_tfgnn_graph_classification_mutag.ipynb) to use a model like this:

```python
graph = graph_sage.GraphSAGEGraphUpdate(
        node_set_names=["atoms"],
        receiver_tag=tfgnn.TARGET,
        hidden_units = 10,      
        units=message_dim,
        name="gsage_layer_" + str(i))(graph)
```
I can repro a similar (but unrelated) error:

> WARNING:tensorflow:Gradients do not exist for variables ['dense_1/kernel:0', 'dense_1/bias:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss` argument?

In this case, the root cause is

```python
  def set_initial_edge_state(edge_set, *, edge_set_name):
    return tf.keras.layers.Dense(edge_dim)(edge_set[tfgnn.HIDDEN_STATE])
```

Which wraps a `Dense` layer around edge features that are never used in the model.  The fix for this tutorial file is to remove this set_initial_edge_state call from `MapFeatures`.

```python
  graph = tfgnn.keras.layers.MapFeatures(
      node_sets_fn=set_initial_node_state)(
          graph)
```

Hope that helps!
